src/amo/components/Addon/index.js

```diff
@@ -280,7 +280,9 @@ export class AddonBase extends React.Component {
     let errorBanner = null;
     if (errorHandler.hasError()) {
       log.error('Captured API Error:', errorHandler.capturedError);
-      if (errorHandler.capturedError.responseStatusCode === 404) {
+      if (errorHandler.capturedError.responseStatusCode === 404 ||
+          errorHandler.capturedError.responseStatusCode === 401
```

Comment:
Two points:

1. nitpick, but could you put 401 first? Just find it easier to scan with incremental counts
2. I think you should add a comment explaining why this is here--even just something like "We send 404 for unauthorised add-ons to prevent unauthorised users from even knowing if the add-on exists." and then a link to the issue